File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys

input = sys.stdin.readline

n = int(input().strip())
arr = list(map(int, input().split()))
target = int(input().strip())

arr.sort()

left, right = 0, n - 1
count = 0

while left < right:
    s = arr[left] + arr[right]
    if s == target:
        count += 1
        left += 1
        right -= 1
    elif s < target:
        left += 1
    else:
        right -= 1

print(count)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
const readline = require('readline');
const rl = readline.createInterface({
    
    input: process.stdin,
    output: process.stdout
}
);

let input = [];
rl.on('line', line => {
    input.push(line.trim());
}).on('close', () => {
    solution();
    process.exit();
});

function solution() {
    const n = Number(input[0]);
    let arr = input[1].split(' ').map(Number).sort((a, b) => a - b);
    const x = Number(input[2]);
    let cnt = 0;
    let left = 0, right = n - 1;

    while(left < right) {
        const sum = arr[left] + arr[right];
        if (sum === x) {
            cnt++;
            left++;
            right--;
        }
        else if(sum < x) {
            left++;
        }
        else {
            right--;
        }
    }
    console.log(cnt);
    
}